warp/Network/Wai/Handler/Warp.hs

```diff
@@ -491,3 +493,10 @@ setGracefulCloseTimeout2 x y = y { settingsGracefulCloseTimeout2 = x }
 -- Since 3.3.5
 getGracefulCloseTimeout2 :: Settings -> Int
 getGracefulCloseTimeout2 = settingsGracefulCloseTimeout2
+
+-- | Getting information of client certificate.
+--
+-- Since 3.3.5
+clientCertificate :: Request -> Maybe CertificateChain
+clientCertificate = fromMaybe (throw (userError "getClientCertificate")) . Vault.lookup getClientCertificateKey . vault
```

Comment:
What's the motivation for using a bottom value here instead of propagating the `Maybe`? If we need to distinguish different cases, I'd rather a specific sum type to represent the different ways things can fail.

Reply:
Oh. It's from copy-and-paste. Does the following code make sense?

```haskell
clientCertificate :: Request -> Maybe CertificateChain
clientCertificate = join . Vault.lookup getClientCertificateKey . vault
```

Reply:
Yup, seems reasonable.

Reply:
Thanks. I pushed this. Can we merge this PR now?